### ISSUE TITLE
[Issue] Error occurs when running a sail.js app using Forever where the controller is written as .coffee instead of .js

### ISSUE THREAD

**Issue by saintc0d3r** [balderdashy/sails#1762]
Synopsis:
----------
When running a sail.js web application project using <code>forever</code> where the controllers are written as <code>.coffee</code> instead of <code>.js</code> files, error occurs & the sail.js app is failing to be lifted.

System Environments:
------------------------------
- Linux Ubuntu 14.04
- Sail.js version: 0.9.16
- Node.js version: 0.10.28

Steps to reproduce:
--------------------------
1. Create a new sail.js app.
2. In the new sail.js app directory, generate a new controller and add an action on the new controller (https://gist.github.com/saintc0d3r/03d5209e4a99c683d1a8).
3. Configure the route.js file so that the '/' would point to the new controller and its action method (https://gist.github.com/saintc0d3r/dcf2e275f92920e89cec).
4. Go to the sail.js app's directory and run <code>sails lift</code> command then browse to <code>localhost:1337</code>. Confirm that the controller returns correct action's result in the browser.
5. Turn off the lifted sail.js app (pressing CTL+C in the terminal where the <code>sails lift</code> command runs ).
6. Create a new controller file inside api/controllers directory where it's name matches with the controller's name created in step 2 but the new file's extension is .coffee.
7. Rewrite the controller js code as coffee script code inside the created new controller (https://gist.github.com/saintc0d3r/0447316428d5091072b7 ).
8. Delete or move the original controller js file. 
9. Do step 4's action and confirm the result.
10. In the sail.js app's root directory, run: <code>npm install</code> command to install required node_modules.
11. Install <code>forever</code> globally (run: <code>sudo npm -g install forever</code> ).
12. In the sail.js app's root directory, lift the app by running this command:
    <code>forever start -ae errors.log app.js --dev --port 1337</code>
13. Open the <code>errors.log</code>. Confirm that there is error message written in the log, along with the error's stacktrace (https://gist.github.com/saintc0d3r/3e9cc5f27b4bb11cc630).
14. Browse to localhost:1337. Confirm that error page is shown.
15. Stop the started failing app by running this command: <code>forever stop app.js</code>.
16. Modify the controller .coffee file by removing all '#' characters.
17. Do step 12 & 13. Confirm that there is different error message written in the log and also its stacktrace (https://gist.github.com/saintc0d3r/af543872691706159aca).
18. Do step 14. Confirm that same error page is shown.
 
Temporary Workaround:
--------------------------------
1. Compile all <code>.coffee</code> files inside api/controllers directory into <code>.js</code> files by running this command within the sail.js app's root directory:
<code>coffee --compile api/controllers</code>
2. Delete/Move the controller coffee files. Before doing this, ensure that you have backup or put the controller coffee files into source control ( e.g. github, bitbucket, etc ).
3. In the sail.js app's root directory, lift the app by running this command:
    <code>forever start -ae errors.log app.js --dev --port 1337</code>

**Comment by mikermcneil** [comment]
Thanks guys-- I believe this was fixed by a later release.  Is anyone using coffeescript still having this issue in v0.11?

**Comment by masumeEbhami** [comment]
i have same problem on sails v0.11 in linux mint

**Comment by mikermcneil** [comment]
Thanks for posting, @saintc0d3r.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
